### PR TITLE
Add `--high-availability` flag for cluster create and update

### PR DESCRIPTION
**Pull request by qrnvttrl**
This is for toggling the HA control plane feature.

### CODE REVIEW

**Comment by mwennrich** [comment]
may use flag `--high-availability-control-plane` instead? It's more clear and leaves us the option to add later other HA flags like `--high-availability-firewall` or so?

File: cmd/cluster.go [hunk 1]
```diff
@@ -965,6 +973,9 @@ func (c *config) updateCluster(args []string) error {
 	if viper.IsSet("logacceptedconns") {
 		clusterFeatures.LogAcceptedConnections = &logAcceptedConnections
 	}
+	if viper.IsSet("high-availability-control-plane") {
+		clusterFeatures.HighAvailability = &highAvailability
```
Comment (Gerrit91) [inline cmd/cluster.go:977]: If this is toggled from false to true, can we also add a prompt like this?

```golang
	if err := genericcli.PromptCustom(&genericcli.PromptConfig{
		Message:         "Enabling the HA control plane feature gate is still a beta feature. Please be aware that you cannot toggle back this feature after it was enabled.",
		ShowAnswers: true,
		Out:             c.out,
	}); err != nil {
		return err
	}
```

I would like to point out that this feature is still `beta`. So we have a bit more time to collect experience in production with it before motivating all customers to enable it.

File: cmd/cluster.go [hunk 2]
```diff
@@ -673,6 +676,17 @@ WARNING: You are going to create a cluster that has no default internet access w
 		}
 	}
 
+	if viper.IsSet("high-availability-control-plane") {
+		scr.ClusterFeatures.HighAvailability = &highAvailability
+		if err := genericcli.PromptCustom(&genericcli.PromptConfig{
```
Comment (Gerrit91) [inline cmd/cluster.go:681]: Check is missing if it the flag is set to `true`.

File: go.mod [hunk 3]
```diff
@@ -8,7 +8,7 @@ require (
 	github.com/dustin/go-humanize v1.0.1
 	github.com/fatih/color v1.17.0
 	github.com/fi-ts/accounting-go v0.10.0
-	github.com/fi-ts/cloud-go v0.28.0
+	github.com/fi-ts/cloud-go v0.28.1-0.20240924075858-f804e87599af
```
Comment (Gerrit91) [inline go.mod:11]: Can now be pinned to `v0.28.2`.

**Review by Gerrit91 — APPROVED** [review]

**Pull request merged**